src/draw_target.rs

```diff
@@ -475,19 +475,37 @@ impl DrawState {
     fn draw_to_term(
         &mut self,
         term: &(impl TermLike + ?Sized),
-        last_line_count: &mut VisualLines,
+        bar_count: &mut VisualLines, // The number of dynamic lines printed at the previous tick
     ) -> io::Result<()> {
         if panicking() {
             return Ok(());
         }
 
+        let term_height = term.height() as usize;
```

Comment:
This first commit has too many logically independent changes going on. In general, I don't think I'm a fan of hoisting up all these variable bindings, further up from where they're actually used.

Reply:
All the changes are variable renaming or movement. The hoisting up is to have a clear definition of what each variable is and relates to the others, especially as some of those variables were defined twice. It is also temporary and here to make the coming changes clearer.

Reply:
Especially for variable renaming and movement, I want those in separate commits so reviewing it is easier. I definitely don't want a whole bunch of variable declarations up front, each variable should be defined shortly before its first usage site.